server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2063,6 +2063,7 @@ private VMTemplateVO updateTemplateOrIso(BaseUpdateTemplateOrIsoCmd cmd) {
             ex.addProxyObject(String.valueOf(id), "templateId");
             throw ex;
         }
+        long oldGuestOSId = template.getGuestOSId();
```

Comment:
moving before this line L#2096 `template = _tmpltDao.createForUpdate(id);` should be ok?